megamek/src/megamek/common/util/TurnTimer.java

```diff
@@ -0,0 +1,119 @@
+ * When zero is reached, the ready() method of the given {@link AbstractPhaseDisplay} is called
+ * to end the users current turn.
+ *
+ */
+public class TurnTimer {
+
+    private Timer timer;
+    private JProgressBar progressBar;
+    private ActionListener listener;
+    private JLabel remaining;
+    private JDialog display;
+    private int timeLimit ;
+
+
+    public TurnTimer(int limit, AbstractPhaseDisplay phaseDisplay) {
+
+        timeLimit = limit;
+        display = new JDialog();
+        progressBar = new JProgressBar(JProgressBar.VERTICAL, 0, limit);
+        progressBar.setValue(limit);
+        progressBar.setString(String.valueOf(timeLimit)+"s");
+
+        listener = new ActionListener() {
+            int counter = limit;
+            public void actionPerformed(ActionEvent ae) {
+                counter--;
+                remaining.setText(counter+" sec");
+                progressBar.setValue(counter);
+                progressBar.setString(String.valueOf(counter)+"s");
+                if (counter<1) {
+                    phaseDisplay.ready();
+                    timer.stop();
+                    display.setVisible(false);
+                    display.dispose();
+                }
+            }
+        };
+
+    }
+
+    public void startTimer() {
+
+        SwingUtilities.invokeLater( new Runnable() {
+            public void run() {
+                remaining = new JLabel(timeLimit+" sec");
+                timer = new Timer(1000, listener);
+                timer.start();
+                display.setLayout(new BorderLayout());
+                display.getContentPane().add(remaining, BorderLayout.NORTH);
+                display.getContentPane().add(progressBar, BorderLayout.CENTER);
+                display.setSize(50,100);
+                display.setAlwaysOnTop(true);
+                display.setModal(false);
+                //display.setUndecorated(true);
+                display.setVisible(true);
+
+            }
+        });
+    }
+    public void stopTimer() {
+        display.setVisible(false);
+        display.dispose();
+        timer.stop();
+    }
+    public boolean isRunning(){
+        return timer.isRunning();
+    }
+    public void resetTimer(){
+        progressBar.setValue(timeLimit);
+        display.setVisible(true);
+
+    }
+
+
+    public static TurnTimer init(AbstractPhaseDisplay phaseDisplay, Client client){
+
+        //check if there should be a turn timer running
+        if (timerShouldStart(client)==true){
+            Option timer = (Option) client.getGame().getOptions().getOption("turn_timer");
+           TurnTimer tt = new TurnTimer(timer.intValue(), phaseDisplay);
+            tt.startTimer();
+            return tt;
+        }
+        return null;
+    }
+
+    /**
+     * Checks if a turn time limit is set in options
+     * limit is only imposed on movement, firing
+     *  @todo: add timer to physical and targeting phase currently it is only in movement and fire
+     */
```

Comment:
@todo is not a JavaDoc tag. This should be outside the JavaDoc block.
```suggestion
    /**
     * Checks if a turn time limit is set in options
     * limit is only imposed on movement, firing
     */
    //TODO: add timer to physical and targeting phase currently it is only in movement and fire
```